_episodes/05b-advanced-containers.md

```diff
@@ -0,0 +1,315 @@
+If you have a "final" version of your data, or want to create an authoritative version
+of your work, you might want to put files into the container permanently. That's
+what we'll cover next.
+
+## Including personal scripts and data in a container
+
+Our next project will be to add our own files to a container - something you
+might want to do if you're sharing a finished analysis or just want to have
+an archived copy of everything. Let's assume that we've finished with our `sum.py`
+script and want to add it to the container itself.
+
+In your shell, you should still be in the `sum` folder in the `docker-intro` folder.
+~~~
+$ pwd
+~~~
+{: .language-bash}
+~~~
+$ /Users/yourname/Desktop/docker-intro/sum
+~~~
+{: .language-bash}
+
+Take a look at the Dockerfile. It looks similar to the one we used before, but
+it has an additional line with the `COPY` keyword.
+
+~~~
+COPY sum.py /home
+~~~
+
+This line will cause Docker to copy the file from your computer into the container's
+file system. Let's build the container like before, but give it a different name:
+
+~~~
+$ docker build -t alice/alpine-sum .
+~~~
+{: .language-bash}
+
+> ## Exercise: Did it work?
+>
+> Can you remember how to run a container interactively? Try that with this one.
+> Once inside, try running the Python script.
+>
+> > ## Solution
+> >
+> > You can start the container interactively like so:
+> > ~~~
+> > $ docker run -it alice/alpine-sum sh
+> > ~~~
+> > {: .language-bash}
+> >
+> > You should be able to run the python command inside the container like this:
+> > ~~~
+> > /# python /home/sum.py
+> > ~~~
+> > {: .language-bash}
+> >
+> {: .solution}
+{: .challenge}
+
+This `COPY` keyword can be used to place your own scripts or own data into a container
+that you want to publish or use as a record. Note that it's not necessarily a good idea
+to put your scripts inside the container if you're constantly changing or editing them.
+Then, referencing the scripts from outside the container is a good idea, as we
+did in the previous section. You also want to think carefully about size -- if you
+run `docker image ls` you'll see the size of each image all the way on the right of
+the screen. The bigger your image becomes, the harder it will be to easily download.
+
+> ## Copying alternatives
+>
+> Another trick for getting your own files into a container is by using the `RUN`
+> keyword and downloading the files from the internet. For example, if your code
+> is in a GitHub repository, you could include this statement in your Dockerfile
+> to download the latest version every time you build the container:
+>
+> ~~~
+> RUN git clone https://github.com/alice/mycode
+> ~~~
+>
+> Similarly, the `wget` command can be used to download any file publicly available
+> on the internet:
+>
+> ~~~
+> RUN wget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.10.0/ncbi-blast-2.10.0+-x64-linux.tar.gz
+> ~~~
+>
+{: callout}
+
+## More fancy `Dockerfile` options
```

Comment:
Maybe this whole section should be a callout to indicate that it is optional for instructors?

Reply:
Hmmm. They could also be optional exercises. Maybe I'll just add optional to the title?

Reply:
Sounds good.